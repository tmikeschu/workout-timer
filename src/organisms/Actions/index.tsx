import * as React from "react";
import { styled } from "theme";
import Button from "atoms/Button";
import { AppMachineContext } from "contexts/machine";
import { createUUID } from "utils";

const Container = styled.div`
  display: flex;
  flex-wrap: wrap;
  padding: 1rem;
  justify-content: space-between;
`;

const Actions: React.FC<{}> = () => {
  const [current, send] = React.useContext(AppMachineContext);
  const addAnnouncement = (): void => {
    const draft = [...current.context.announcementTimes];
    draft.push({ time: 0, interval: false, id: createUUID() });
    send({
      type: "SET_ANNOUNCEMENT_TIMES",
      payload: { announcementTimes: draft }
    });
  };
  const fresh = current.context.initialTime === current.context.currentTime;
  const start = (): void => {
    window.speechSynthesis.speak(new SpeechSynthesisUtterance("Let's go"));
    send("START");
  };

  return (
    <Container>
      <Button disabled={current.matches("running")} onClick={start}>
        Start
      </Button>
      <Button
        disabled={!current.matches("running")}
        onClick={(): void => {
          send("STOP");
        }}
      >
        Stop
      </Button>

      <Button
        disabled={fresh || current.matches("running")}
        onClick={(): void => {
          send("RESET");
        }}
      >
        Reset
      </Button>
      <Button
        onClick={addAnnouncement}
        disabled={!fresh || current.matches("running")}
      >
        Add Announcement
      </Button>
    </Container>
  );
};

export default Actions;
