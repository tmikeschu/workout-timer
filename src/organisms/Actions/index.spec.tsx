import React from "react";
import { render } from "test";
import Actions from "./";

describe("Actions", () => {
  it("renders correctly", () => {
    const tree = render(<Actions />).container;
    expect(tree).toMatchSnapshot();
  });

  it("shows announcement buttons", () => {
    const tree = render(<Actions />);
    tree.getByTestId("show-more").click();

    expect(tree.container).toMatchSnapshot();
  });

  it("shows save options", () => {
    localStorage.setItem(
      "timerConfigs",
      JSON.stringify({
        nice: [
          {
            message: "hello",
            id: "aoeu",
            time: 1,
            interval: false,
          },
        ],
      })
    );
    const tree = render(<Actions />);
    tree.getByTestId("show-more").click();
    tree.getByTestId("add-announcement").click();
    tree.getByTestId("save-announcements").click();

    expect(tree.container).toMatchSnapshot();
  });
});
