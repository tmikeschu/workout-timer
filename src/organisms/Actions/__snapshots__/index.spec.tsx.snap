// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions renders correctly 1`] = `
<div>
  <div
    class="css-vcirm2"
  >
    <button
      class="css-twz5y6"
    >
      Start
    </button>
    <button
      class="css-twz5y6"
      disabled=""
    >
      Stop
    </button>
    <button
      class="css-twz5y6"
      disabled=""
    >
      Reset
    </button>
    <button
      class="css-twz5y6"
      data-testid="Actions__addAnnouncement"
    >
      Add Announcement
    </button>
  </div>
</div>
`;
